include/aie/Dialect/AIE/IR/AIETargetModel.h

```diff
@@ -54,7 +54,7 @@ using TileID = struct TileID {
 
 class AIETargetModel {
 public:
-  AIETargetModel() = default;
+  AIETargetModel(AIEDevice device) : device(device) {}
```

Comment:
I think this is not a good idea...  My intention with the TargetModel is that it's a singleton pattern, which is independent of the MLIR.  an AIEDevice is the device *of a particular design*.  And a Device is not actually necessary, only a container object that implements the right interface.  It's unclear to me why you need to walk from the TargetModel back to a device anyway?

Reply:
It was convenient for wrapping/unwrapping to keep only the enum around, and this helped with that. https://github.com/Xilinx/mlir-aie/pull/1500/files#diff-3f51e6951752378031f1f22e2616b9d3c77de3a54fcbb31cdc6ab6bd34d109dbR18-R24. But I can probably refactor to avoid the change to `TargetModel`